File: src/main/java/org/zenbot/szolnok/timetable/batch/step/bus/processor/EndBusStopSelectorItemProcessorHelper.java
package org.zenbot.szolnok.timetable.batch.step.bus.processor;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;
import org.jsoup.select.Elements;
import org.springframework.stereotype.Component;
import org.zenbot.szolnok.timetable.configuration.properties.TimetableSelectorProperties;

@Component
public class EndBusStopSelectorItemProcessorHelper {

    private final StringCleaner stringCleaner;

    public EndBusStopSelectorItemProcessorHelper(StringCleaner stringCleaner) {
        this.stringCleaner = stringCleaner;
    }

    public String getEndBusStop(Document htmlDocument, TimetableSelectorProperties selectorProperties) {
        Elements stationsTable = htmlDocument.select(selectorProperties.getBusStopsSelector());
        Elements rows = stationsTable.select(selectorProperties.getTableRowSelector());
        int indexOfEndBusStop = rows.size() - 2;
        Element lastRow = rows.get(indexOfEndBusStop);
        Elements lastRowColumns = lastRow.select(selectorProperties.getTableColumnSelector());
        Element lastColumn = lastRowColumns.get(2);
        String to = lastColumn.text();
        return  stringCleaner.clean(to);
    }
}

File: src/main/kotlin/org/zenbot/szolnok/timetable/batch/step/bus/processor/EndBusStopSelectorItemProcessorHelper.kt
package org.zenbot.szolnok.timetable.batch.step.bus.processor

import org.jsoup.nodes.Document
import org.springframework.stereotype.Component
import org.zenbot.szolnok.timetable.configuration.properties.TimetableSelectorProperties

@Component
class EndBusStopSelectorItemProcessorHelper(private val stringCleaner: StringCleaner) {

    fun getEndBusStop(htmlDocument: Document, selectorProperties: TimetableSelectorProperties): String {
        val stationsTable = htmlDocument.select(selectorProperties.busStopsSelector)
        val rows = stationsTable.select(selectorProperties.tableRowSelector)
        val indexOfEndBusStop = rows.size - 2
        val lastRow = rows[indexOfEndBusStop]
        val lastRowColumns = lastRow.select(selectorProperties.tableColumnSelector)
        val lastColumn = lastRowColumns[2]
        val to = lastColumn.text()
        return stringCleaner.clean(to)
    }
}
